### ISSUE TITLE
Support grid search in parallel on GPUs

### ISSUE THREAD

**Issue by ottonemo** [dnouri/skorch#95]
It would be nice to have `n_jobs=2` on a system with 2 GPUs and both jobs are dispatched to one of the GPUs.

**Comment by ottonemo** [comment]
The new device handling changed the WIP solution significantly so we should re-think the way this can be done before 0.2.0. Therefore moving the target to 0.3.0.

**Comment by ottonemo** [comment]
An explanation how to train in parallel on multiple GPUs (grid or random search) using Dask was added in #241.
